pkg/util/constants.go

```diff
@@ -98,6 +100,10 @@ func metamorphicEligible() bool {
 	if !buildutil.CrdbTestBuild {
 		return false
 	}
+	// Bail out if we are running a non-test binary which also happens to be non-cockroach, e.g., roachprod, roachtest.
+	if !bazel.InBazelTest() && !strings.Contains(build.GetInfo().BinaryName, "cockroach") {
```

Comment:
Can you elaborate a bit on this predicate? Why do we need the `!bazel.InBazelTest()` part?